osu.Game/Skinning/LegacySkinTransformer.cs

```diff
@@ -1,28 +1,62 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System.Linq;
 using osu.Framework.Audio.Sample;
+using osu.Framework.Graphics;
 using osu.Game.Audio;
 using osu.Game.Rulesets.Objects.Legacy;
+using osuTK;
 using static osu.Game.Skinning.SkinConfiguration;
 
 namespace osu.Game.Skinning
 {
-    /// <summary>
-    /// Transformer used to handle support of legacy features for individual rulesets.
-    /// </summary>
-    public abstract class LegacySkinTransformer : SkinTransformer
+    public class LegacySkinTransformer : SkinTransformer
     {
         /// <summary>
         /// Whether the skin being transformed is able to provide legacy resources for the ruleset.
         /// </summary>
         public virtual bool IsProvidingLegacyResources => this.HasFont(LegacyFont.Combo);
 
-        protected LegacySkinTransformer(ISkin skin)
+        public LegacySkinTransformer(ISkin skin)
             : base(skin)
         {
         }
 
+        public override Drawable? GetDrawableComponent(ISkinComponentLookup lookup)
+        {
+            switch (lookup)
+            {
+                case SkinComponentsContainerLookup containerLookup:
+                    switch (containerLookup.Target)
+                    {
+                        case SkinComponentsContainerLookup.TargetArea.MainHUDComponents when containerLookup.Ruleset != null:
+                            var rulesetHUDComponents = base.GetDrawableComponent(lookup);
+
+                            rulesetHUDComponents ??= new DefaultSkinComponentsContainer(container =>
+                            {
+                                var combo = container.OfType<LegacyComboCounter>().FirstOrDefault();
+
+                                if (combo != null)
+                                {
+                                    combo.Anchor = Anchor.BottomLeft;
+                                    combo.Origin = Anchor.BottomLeft;
+                                    combo.Scale = new Vector2(1.28f);
+                                }
+                            })
+                            {
+                                new LegacyComboCounter()
+                            };
```

Comment:
> This is so confusing to read.
> 
> * The ctor argument is a callback that gets executed on load complete.
> * Then there is some sort of weird initialiser syntax that is _really_ just shorthand for an `.Add()` call (which I realised after staring this for 2 minutes wondering how this is even valid code).
> 
> So the first executes _after_ the second, plus the added part of the weird syntax inside.
> 
> Can we not? Can the add just be a plain add?

I'm following how every skin implementation initialises the default layout of its components. In this specific case, I can avoid using `DefaultSkinComponentsContainer` because I'm not relying on the layout of another component in the container like `LegacySkin` does for example, but I didn't pay much attention because I would expect you to be equally confused when looking at any skin implementation's handling of target lookups (e.g. `LegacySkin.GetDrawableComponent(MainHUDComponents)`).

I'll simplify it anyway.
 
> Also, just to confirm that I'm understanding correctly, because these systems are confusing - this is only handling the default skin case, correct? If there is _anything_ customised on the user's side the base call will kick in, rendering all of this logic no-op?

Correct, that's also how it works in `LegacySkin`/`ArgonSkin` etc.